pkg/ottl/contexts/internal/ottlcommon/span.go

```diff
@@ -339,12 +340,27 @@ func accessAttributesKey[K SpanContext](mapKey *string) ottl.StandardGetSetter[K
 			return GetMapValue(tCtx.GetSpan().Attributes(), *mapKey), nil
 		},
 		Setter: func(ctx context.Context, tCtx K, val interface{}) error {
+			if ok := checkValuesHomogeneous(val); !ok {
+				return fmt.Errorf("arrays in attribute values are required to be homogeneous in span")
+			}
 			SetMapValue(tCtx.GetSpan().Attributes(), *mapKey, val)
 			return nil
 		},
 	}
 }
 
+func checkValuesHomogeneous(val interface{}) bool {
+	if items, ok := val.([]interface{}); ok && len(items) > 0 {
+		lasttype := reflect.TypeOf(items[0])
+		for _, item := range items {
+			if reflect.TypeOf(item) != lasttype {
+				return false
+			}
+		}
```

Comment:
To keep this efficient, I would change this to:

```suggestion
		for i := 1; i < len(items); i++ {
			if reflect.TypeOf(items[i]) != lasttype {
				return false
			}
		} 
```